cmd/podman/common/completion.go

```diff
@@ -664,6 +664,30 @@ func AutocompleteSystemConnections(cmd *cobra.Command, args []string, toComplete
 	return suggestions, cobra.ShellCompDirectiveNoFileComp
 }
 
+// AutocompleteScp returns a list of connections, images, or both, depending on the amount of arguments
+func AutocompleteScp(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if !validCurrentCmdLine(cmd, args, toComplete) {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+	connectionSuggestions, _ := AutocompleteSystemConnections(cmd, args, toComplete)
+	imageSuggestions, _ := getImages(cmd, toComplete)
+
+	switch len(args) {
+	case 0:
+		totalSuggestions := append(connectionSuggestions, imageSuggestions...)
+		return totalSuggestions, cobra.ShellCompDirectiveNoFileComp
+	case 1:
+		if strings.Contains(args[0], "::") {
+			if len(strings.Split(args[0], "::")[1]) > 0 {
```

Comment:
Probably late-night brain, but I'm not sure what the `[1]` does here?  Hopefully, you'll teach me some new golang magic incantation!  ;^)

Reply:
`strings.Split` is just an array containing the connection name in index 0 and the image name (if given) in index 1. Here I was checking if the user gave an image name yet which means we should not suggest anything else if they did

Reply:
@cdoern Ah, thanks, neat trick!  However, I just ran a test in the Golang sandbox and it panics if there's no colons in the string in args[0], but works fine if there is.  I think you need a check to make sure it's in there unless you 1,000 percent positive of it's presence.

```
package main

import "fmt"
import "strings"

func main() {
	fmt.Println("Hello, 世界")
		
        if len(strings.Split("TomSweeney", "::")[1]) > 0 {
            fmt.Println("IN if")
        }
	fmt.Println("Done!")
}


Run:
Hello, 世界
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
main.main()
	/tmp/sandbox584709085/prog.go:11 +0x19d

Program exited.
```

Reply:
what about `strings.SplitN()`with N = 2? I think this ensures there will always be two entries

Reply:
This is belching for me too:

```
// You can edit this code!
// Click here and start typing.
package main

import "fmt"
import "strings"

func main() {
	fmt.Println("Hello, 世界")
		
        if len(strings.SplitN("TomSweeney", "::", 2)[1]) > 0 {
            fmt.Println("IN if")
        }
	fmt.Println("Done!")
}
```

Reply:
The code is different now. Anyway he had `if strings.Contains(args[0], "::") {` before so it was safe to to.